src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc

```diff
@@ -0,0 +1,289 @@
+const uint32_t kFunctionId = 2;
+const uint32_t kAppId = 5u;
+}  // namespace
+
+class UpdateTurnListRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  UpdateTurnListRequestTest()
+      : mock_message_helper_(am::MockMessageHelper::message_helper_mock())
+      , command_msg_(CreateMessage(smart_objects::SmartType_Map))
+      , command_(CreateCommand<UpdateTurnListRequest>(command_msg_)) {
+    (*command_msg_)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+    (*command_msg_)[am::strings::params][am::strings::function_id] =
+        kFunctionId;
+  }
+
+  am::MockMessageHelper* mock_message_helper_;
+  MessageSharedPtr command_msg_;
+  ::utils::SharedPtr<UpdateTurnListRequest> command_;
+};
+
+TEST_F(UpdateTurnListRequestTest, Run_ApplicationIsNotRegistered_UNSUCCESS) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(ApplicationSharedPtr()));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest, Run_InvalidNavigationText_UNSUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = "invalid_navigation_text\t\n";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest, Run_InvalidTurnIcon_UNSUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = "valid_navigation_text";
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "invalid_turn_icon\t\n";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest,
+       Run_ProcessSoftButtonsNotSucceeded_UNSUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::navigation_text] = "valid_navigation_text";
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "valid_turn_icon";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@LevchenkoS Do you really dereference a NULL pointer? If so this means undefined behavior! Under Linux - a crash!